src/main/java/io/cdap/plugin/gcp/spanner/common/SpannerUtil.java

```diff
@@ -138,6 +145,46 @@ public static String convertSchemaToCreateStatement(String tableName, String pri
     return createStmt.toString();
   }
 
+  private static String getArrayType(Schema componentSchema) {
+    Schema unionSchema = componentSchema.getUnionSchema(0);
```

Comment:
Shouldn't assume the component schema is a union. instead, do:

```
componentSchema = componentSchema.isNullable() ? componentSchema.getNonNullable() : componentSchema;
```

Reply:
done